tests/conftest.py

```diff
@@ -220,6 +220,44 @@ def _create_profile(name, **kwargs):
     return _create_profile
 
 
+@pytest.fixture
+def create_config_instance(create_empty_config_instance, create_profile) -> Config:
+    """Create a temporary configuration instance with one profile.
+    This creates a temporary directory with a clean `.aiida` folder and basic configuration file.
+    The currently loaded configuration and profile are stored in memory,
+    and are automatically restored at the end of this context manager.
+    This fixture differs from `create_empty_config_instance` in that here a profile will be created and set as default.
+    The defaults of the profile can be overridden in the callable, as well as whether it should be set as default.
+    This fixture should probably be used for most tests that assume a basically configured setup.
```

Comment:
Ok, here my suggestion:

```suggestion
    
    This fixture builds on the `create_empty_config_instance`, adds one profile and set it as default.
    The defaults of the profile can be overridden in the callable, as well as whether it should be set as default.
    
    Example::
    
       ...
    
    As with `create_empty_config_instance`, the currently loaded configuration and profile are stored in memory,
    and are automatically restored at the end of this context manager.
    This fixture should be used by tests that modify aspects of the AiiDA configuration or profile
    and require a preconfigured profile.
```

Reply:
P.S. It's not very explicit in the code how the db backend of the profile is chosen.

It is important that the backend be the same one of the test profile - otherwise any operations on the DB will fail.

Reply:
It's intentionally not mentioned because this fixture should not actually be used to "run" the profile. It should merely be used to test functionality that manipulates the content of the config folder, such as the `profile.json`. So I would propose just to change the docsting to make this more prominent. Then it is not a problem that the backend may be incorrect

Reply:
Ok, yes then this should be stated clearly in the docstring of the fixture (probably also in the one for the empty config).

I was a bit confused since I saw the `load_profile` - but I guess that does not actually establish a database connection here (?)

Reply:
> I was a bit confused since I saw the load_profile - but I guess that does not actually establish a database connection here

I actually added that on top of @sphuber's original function, because some tests failed since `config.current_profile` returned `None`, e.g.

```python
_________________________________________________ TestVerdiConfig.test_config_remove_option _________________________________________________

self = <tests.cmdline.commands.test_config.TestVerdiConfig object at 0x7f90129bafd0>
run_cli_command = <function run_cli_command.<locals>._run_cli_command at 0x7f9012b857a0>

    def test_config_remove_option(self, run_cli_command):
        """Test the `verdi config set --remove` command when removing an option value."""
        config = get_config()
    
        option_name = 'caching.disabled_for'
>       config.set_option(option_name, ['x', 'y'], scope=config.current_profile.name)
E       AttributeError: 'NoneType' object has no attribute 'name'

tests/cmdline/commands/test_config.py:125: AttributeError
```

Reply:
correct `load_profile` just sets the global variable with the contents of the JSON settings, but it does not actually load the backend

Reply:
ok updated in a9189eccb797affc59399998da17a900595bd1c3